great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -177,22 +177,76 @@ def get_checkpoint(
         return checkpoint_config
 
     def add_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+        """Persist a stand-alone Checkpoint object.
+
+        Args:
+            checkpoint: The Checkpoint to persist.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+
+        Raises:
+            CheckpointError: If a Checkpoint with the given name already exists.
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
+
+        try:
+            checkpoint_ref = self.add(key=key, value=checkpoint.get_config())  # type: ignore[func-returns-value]
+        except gx_exceptions.StoreBackendError:
+            raise gx_exceptions.CheckpointError(
+                f"A Checkpoint named {checkpoint.name} already exists."
+            )
+
+        if isinstance(checkpoint_ref, GXCloudIDAwareRef):
+            cloud_id = checkpoint_ref.cloud_id
+            checkpoint.ge_cloud_id = uuid.UUID(cloud_id)  # type: ignore[misc]
+
+        return checkpoint
 
     def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+        """Use a stand-alone Checkpoint object to update a persisted value.
+
+        Args:
+            checkpoint: The Checkpoint to use for updating.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+
+        Raises:
+            CheckpointNotFoundError: If a Checkpoint with the given name does not exist in the store.
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        if not self.has_key(key):
+
+        try:
+            checkpoint_ref = self.update(key=key, value=checkpoint.get_config())  # type: ignore[func-returns-value]
+        except gx_exceptions.StoreBackendError:
             raise gx_exceptions.CheckpointNotFoundError(
-                f"Could not find a Checkpoint named {checkpoint.name}"
+                f"Could not find an existing Checkpoint named {checkpoint.name}."
             )
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
+
+        if isinstance(checkpoint_ref, GXCloudIDAwareRef):
+            cloud_id = checkpoint_ref.cloud_id
+            checkpoint.ge_cloud_id = uuid.UUID(cloud_id)  # type: ignore[misc]
```

Comment:
What's the misc issue here?

Reply:
This is a bug - `ge_cloud_id` is read-only. I've changed this so the value is actually updated